platforms/jvm/code-quality/src/main/java/org/gradle/api/plugins/quality/internal/CheckstyleInvoker.java

```diff
@@ -0,0 +1,287 @@
+        boolean showViolations = parameters.getShowViolations().get();
+        int maxErrors = parameters.getMaxErrors().get();
+        int maxWarnings = parameters.getMaxWarnings().get();
+        Map<String, Object> configProperties = parameters.getConfigProperties().get();
+        boolean ignoreFailures = parameters.getIgnoreFailures().get();
+        RegularFile config = parameters.getConfig().get();
+        File configDir = parameters.getConfigDirectory().getAsFile().get();
+        boolean isXmlRequired = parameters.getIsXmlRequired().get();
+        boolean isHtmlRequired = parameters.getIsHtmlRequired().get();
+        boolean isSarifRequired = parameters.getIsSarifRequired().get();
+        File xmlOutputLocation = getXmlOutputLocation(isXmlRequired, isHtmlRequired);
+        Property<String> stylesheetString = parameters.getStylesheetString();
+        File htmlOutputLocation = parameters.getHtmlOutputLocation().getAsFile().getOrNull();
+        File sarifOutputLocation = parameters.getSarifOutputLocation().getAsFile().getOrNull();
+        VersionNumber currentToolVersion = determineCheckstyleVersion(Thread.currentThread().getContextClassLoader());
+
+        // User provided their own config_loc
+        Object userProvidedConfigLoc = configProperties.get(CONFIG_LOC_PROPERTY);
+        if (userProvidedConfigLoc != null) {
+            throw new InvalidUserDataException("Cannot add config_loc to checkstyle.configProperties. Please configure the configDirectory on the checkstyle task instead.");
+        }
+
+        if (isSarifRequired && !isSarifSupported(currentToolVersion)) {
+            assertUnsupportedReportFormatSARIF(currentToolVersion);
+        }
+
+        try {
+            ant.taskdef(ImmutableMap.of(
+                "name", "checkstyle",
+                "classname", "com.puppycrawl.tools.checkstyle.CheckStyleTask"
+            ));
+        } catch (RuntimeException ignore) {
+            ant.taskdef(ImmutableMap.of(
+                "name", "checkstyle",
+                "classname", "com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask"
+            ));
+        }
+
+        try {
+            ant.invokeMethod("checkstyle",
+                ImmutableMap.of(
+                    "config", config.getAsFile(),
+                    "failOnViolation", false,
+                    "maxErrors", maxErrors,
+                    "maxWarnings", maxWarnings,
+                    "failureProperty", FAILURE_PROPERTY_NAME
+                ), () -> {
+                    source.addToAntBuilder(ant, "fileset", FileCollection.AntType.FileSet);
+
+                    if (showViolations) {
+                        ant.invokeMethod("formatter", ImmutableMap.of("type", "plain", "useFile", false));
+                    }
+
+                    if (isXmlRequired || isHtmlRequired) {
+                        ant.invokeMethod("formatter", ImmutableMap.of("type", "xml", "toFile", checkNotNull(xmlOutputLocation)));
```

Comment:
This should have a message as well, since it can be `null` here and the user should get a nice message about that.

Reply:
Add a message